filebeat/crawler/registrar.go

```diff
@@ -74,7 +77,7 @@ func (r *Registrar) LoadState() {
 		defer existing.Close()
 		logp.Info("Loading registrar data from %s", r.registryFile)
 		decoder := json.NewDecoder(existing)
-		decoder.Decode(&r.State)
+		decoder.Decode(r.getFullState())
```

Comment:
Consider fully protecting LoadState by mutex, as internal state is completely overwritten. getFullState() will make decoder.Decode overwrite internal state without holding the mutex